src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -1843,6 +1857,362 @@ ReturnCode_t DomainParticipantImpl::unregister_type(
+        {
+            if (service_it.second->service_type_in_use(service_type_name))
+            {
+                EPROSIMA_LOG_ERROR(PARTICIPANT, "Service Type " << service_type_name << " is in use by service " <<
+                        service_it.first);
+                return RETCODE_PRECONDITION_NOT_MET;
+            }
+        }
+    }
+
+    // Unregister request/reply types
+    ReturnCode_t ret_code;
+    ret_code = unregister_type(service_type_name + "_Request");
+
+    if (RETCODE_OK != ret_code)
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Error unregistering Request Type for Service Type " << service_type_name);
+        return ret_code;
+    }
+
+    ret_code = unregister_type(service_type_name + "_Reply");
+
+    if (RETCODE_OK != ret_code)
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Error unregistering Reply Type for Service Type " << service_type_name);
+
+        // Request topic type was unregistered, so register it again to avoid leaving the participant in an inconsistent state
+        register_type(t.request_type(), service_type_name + "_Request");
+
+        return ret_code;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(mtx_service_types_);
+        service_types_.erase(service_type_name);
+    }
+
+    return RETCODE_OK;
+}
+
+rpc::Service* DomainParticipantImpl::create_service(
+        const std::string& service_name,
+        const std::string& service_type_name)
+{
+    if (service_name.empty())
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Service name cannot be empty.");
+        return nullptr;
+    }
+
+    if (service_type_name.empty())
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Service type name cannot be empty.");
+        return nullptr;
+    }
+
+    // Check if the service has already been created
+    {
+        std::lock_guard<std::mutex> lock(mtx_services_);
+        auto it = services_.find(service_name);
+
+        if (it != services_.end())
+        {
+            EPROSIMA_LOG_ERROR(PARTICIPANT, "Service with name '" << service_name << "' already exists.");
+            return nullptr;
+        }
+    }
+
+    // Check if the request/reply content filter factory is registered. If not, register it.
+    IContentFilterFactory* factory = find_content_filter_factory("REQUEST_REPLY_CONTENT_FILTER");
+
+    if (!factory)
+    {
+        factory = new rpc::RequestReplyContentFilterFactory();
+        ReturnCode_t ret_code = register_content_filter_factory("REQUEST_REPLY_CONTENT_FILTER", factory);
+
+        if (RETCODE_OK != ret_code)
+        {
+            EPROSIMA_LOG_ERROR(PARTICIPANT, "Error registering Request/Reply Content Filter Factory.");
+            return nullptr;
+        }
+    }
+
+    // Create and store the service (Internally, this will create the required DDS Request/Reply Topics))
+    rpc::ServiceImpl* service(nullptr);
+    
+    try
+    {
+        service = new rpc::ServiceImpl(service_name, service_type_name, this);  
```

Comment:
... this means we should create them here, and pass them to the service constructor, which would then pass them to the requester / replier constructors